hedera-node/src/main/java/com/hedera/services/ledger/TransactionalLedger.java

```diff
@@ -234,21 +197,79 @@ public void create(K id) {
 		assertIsCreatable(id);
 
 		changes.put(id, new EnumMap<>(propertyType));
+		createdKeys.add(id);
 	}
 
 	@Override
 	public void destroy(K id) {
 		throwIfNotInTxn();
 
 		deadEntities.add(id);
+		perishedKeys.add(id);
 	}
 
 	boolean isInTransaction() {
 		return isInTransaction;
 	}
 
+	String changeSetSoFar() {
+		StringBuilder desc = new StringBuilder("{");
+		AtomicBoolean isFirstChange = new AtomicBoolean(true);
+		changes.entrySet().forEach(change -> {
+			if (!isFirstChange.get()) {
+				desc.append(", ");
+			}
+			K id = change.getKey();
+			var accountInDeadAccounts = deadEntities.contains(id) ? "*DEAD* " : "";
+			var accountNotInDeadAccounts = deadEntities.contains(id) ? "*NEW -> DEAD* " : "*NEW* ";
+			var prefix = entities.contains(id)
+					? accountInDeadAccounts
+					: accountNotInDeadAccounts;
+			desc.append(prefix)
+					.append(keyToString.orElse(EntityIdUtils::readableId).apply(id))
+					.append(": [");
+			desc.append(
+					change.getValue().entrySet().stream()
+							.map(entry -> String.format("%s -> %s", entry.getKey(), readableProperty(entry.getValue())))
+							.collect(joining(", ")));
+			desc.append("]");
+			isFirstChange.set(false);
+		});
+		deadEntities.stream()
+				.filter(id -> !changes.containsKey(id))
+				.forEach(id -> {
+					if (!isFirstChange.get()) {
+						desc.append(", ");
+					}
+					desc.append("*DEAD* ").append(readableId(id));
+					isFirstChange.set(false);
+				});
```

Comment:
complained by Codecov, should add a unit test to cover this part. Also, I think here we can use a simple for loop instead of `stream()`